core/src/main/java/org/elasticsearch/search/aggregations/metrics/scripted/ScriptedMetricAggregatorBuilder.java

```diff
@@ -47,6 +63,38 @@ public ScriptedMetricAggregatorBuilder(String name) {
     }
 
     /**
+     * Read from a stream.
+     */
+    public ScriptedMetricAggregatorBuilder(StreamInput in) throws IOException {
+        super(in, InternalScriptedMetric.TYPE);
+        initScript = in.readOptionalStreamable(Script.SUPPLIER);
+        mapScript = in.readOptionalStreamable(Script.SUPPLIER);
+        combineScript = in.readOptionalStreamable(Script.SUPPLIER);
+        reduceScript = in.readOptionalStreamable(Script.SUPPLIER);
```

Comment:
Another PR, yeah. I wanted to do it here but didn't want to make this one big. Honestly we might be better off making Script a "normal" writeable rather than a Streamable.